include/json/value.h

```diff
@@ -89,9 +76,17 @@ class JSON_API LogicError : public Exception {
 #endif
 
 /// used internally
-JSONCPP_NORETURN void throwRuntimeError(String const& msg);
-/// used internally
-JSONCPP_NORETURN void throwLogicError(String const& msg);
+#if JSON_USE_EXCEPTION
+#define THROW_RUNTIME_ERROR(msg_ref)                                           \
+  { throw RuntimeError(msg_ref); }
+#define THROW_LOGIC_ERROR(msg_ref)                                             \
+  { throw LogicError(msg_ref); }
+#else
+#define THROW_RUNTIME_ERROR(msg_ref)                                           \
+  { return {}; }
```

Comment:
I guess there's no logging facility, so if !JSON_USE_EXCEPTION the message gets swallowed.  Too bad.